### ISSUE TITLE
Plugin should be the only place to manage Executables

### ISSUE THREAD

**Issue by ReinitiaIized** [Bleu-Pigs/studio-plugin#3]
Using a GUI, the Studio Plugin will be the single location to manage Executables for the time being.

**Comment by o-y** [comment]
Ideally the user should be able to overwrite existing modules

**Constraints**
If users can overwrite modules should we keep a history of file revisions so that users can rollback content (like that of the website)?

**Comment by o-y** [comment]
**Managing assets**
- Should users be able to set the name and description of assets?
- Should users be able to set owners and viewers of assets?
  - Owner: Can edit and overwrite the existing asset on Roblox
  - Viewers: Can view the asset but cannot overwrite the asset 

**Constraints**
- The names and descriptions of assets are publically facing, does Roblox preemptively filter out inappropriate/rule breaking content without warning the host? (in this instance an automated Roblox account).
- Assets should always remain private on Roblox.

**Comment by ReinitiaIized** [comment]
> If users can overwrite modules should we keep a history of file revisions so that users can rollback content?

Yes, and this is already implemented for us through asset version history. It shouldn't be difficult to pull down this information and revert accordingly

> Should users be able to set the name and description of assets?

Definitely. When an executable is created, it will become its own unique asset and not shared with any others. However, if we decide assets should stay invisible on ROBLOX, then I don't see much of a point in allowing descriptions as naming should suffice for personal referencing.

> Should users be able to set owners and viewers of assets? 

Although this should be definitely considered, this is currently out of scope for the moment. We want to get MVP functionality going first, then implement Quality of Life improvements once we have MVP established. I'll make a separate issue for this with lower priority.

> The names and descriptions of assets are publically facing, does Roblox preemptively filter out inappropriate/rule breaking content without warning the host? (in this instance an automated Roblox account).

We should be able to use [Chat:FilterStringForBroadcast](https://developer.roblox.com/en-us/api-reference/function/Chat/FilterStringForBroadcast) to get a expectation of what won't pass as it's similar to the website filter.

**Comment by o-y** [comment]
> We should be able to use Chat:FilterStringForBroadcast to get a expectation of what won't pass as it's similar to the website filter.

This would be validation on the client. Does Roblox provide an API that may be used from the web server as well?

**Comment by ReinitiaIized** [comment]
We could possibly attempt to see what backend API is called via Wireshark/Fiddler, otherwise this will be our best option.
